raiden_contracts/contract_manager.py

```diff
@@ -103,21 +125,83 @@ def contracts_deployed_path(
     return data_path.joinpath(f'deployment_{"services_" if services else ""}{chain_name}.json')
 
 
+@deprecated(reason='Use get_contract_deployment_info()')
 def get_contracts_deployed(
         chain_id: int,
         version: Optional[str] = None,
         services: bool = False,
-):
+) -> Dict:
     """Reads the deployment data."""
-    deployment_file_path = contracts_deployed_path(
+    return get_contracts_deployment_info(
         chain_id=chain_id,
         version=version,
-        services=services,
+        module='services' if services else 'raiden',
     )
 
-    try:
-        with deployment_file_path.open() as deployment_file:
-            deployment_data = json.load(deployment_file)
-    except (JSONDecodeError, UnicodeDecodeError, FileNotFoundError) as ex:
-        raise ValueError(f'Cannot load deployment data file: {ex}') from ex
+
+def merge_deployment_data(dict1: Dict, dict2: Dict) -> Dict:
+    """ Take contents of two deployment JSON files and merge them
+
+    The dictionary under 'contracts' key will be merged. The 'contracts'
+    contents from different JSON files must not overlap. The contents
+    under other keys must be identical.
+    """
+    if not dict1:
+        return dict2
+    if not dict2:
+        return dict1
+    result = {}
+    for k1, v1 in dict1.items():
+        if k1 == 'contracts':
+            v: DeployedContracts = deepcopy(v1)
+            # If keys overlap, we would be overwriing some contents away.
+            assert not v.keys() & dict2['contracts'].keys()
+            v.update(dict2['contracts'])
+            result['contracts'] = v
+        else:
+            assert dict2[k1] == v1
+            result[k1] = v1
+    return result
+
+
+def get_contracts_deployment_info(
+        chain_id: int,
+        version: Optional[str] = None,
+        module: str = 'all',
```

Comment:
I had commented about it before but it somehow got lots. Can you change this module to an Enum? It will be much better typed, easier to check and a better interface?

Reply:
There you go.  I defined and used an Enum.